Load only on Drupal 7 sites

SebCorbin opened the pull request:
The test for Drupal.ajax fix loading on old Drupal 6 sites (like localize.drupal.org) where behaviors have a different structure.

On localize.drupal.org I get a `Uncaught TypeError: object is not a function` in

``` JavaScript
Drupal.attachBehaviors = function(context) {
  context = context || document;
  // Execute all of them.
  jQuery.each(Drupal.behaviors, function() {
    this(context);
  });
};
```

SebCorbin committed:
Load only on Drupal 7 sites

The test for Drupal.ajax fix loading on old Drupal 6 sites (like localize.drupal.org) where behaviors have a different structure.

On localize.drupal.org I get a `Uncaught TypeError: object is not a function` in

```JavaScript
Drupal.attachBehaviors = function(context) {
  context = context || document;
  // Execute all of them.
  jQuery.each(Drupal.behaviors, function() {
    this(context);
  });
};
```

sun commented on src/js/init.js:
```diff
@@ -22,7 +22,10 @@ if (window.jQuery !== undefined && window.jQuery.fn.jquery >= '1.4.4' && window.
 jQuery(document).ready(function () {
-  Drupal.attachBehaviors(this);
+  // If we are on a Drupal 7 site (i.e: not localize.drupal.org)
+  if (Drupal.ajax) {
```
ajax.js is not necessarily loaded, so Drupal.ajax may not exist, despite being on a D7 site.

SebCorbin committed:
Fix Drupal.ajax may not be loaded on some pages

Drupal.ajax is present only if ajax.js is. Switch to test for Drupal.detachBehaviors, which is not present in Drupal 6
